Makefile

```
@@ -72,12 +76,12 @@ unittests:
 	@echo "Running unit tests"
 	@tox -e unit -- ${UNIT_ARGS}
 
-functional:
-	@echo "Executing functional tests(charm will be build during functional tests)"
-	@tox -e func -- ${FUNC_ARGS}
+functional: 
```

It seems that a `build` target is needed here.
```suggestion
functional: build
```

---

No it's not. The pytest-operator will handle charm build for us.

---

Noted. Then could you please remove the trailing blank space at the end?
```suggestion
functional:
```